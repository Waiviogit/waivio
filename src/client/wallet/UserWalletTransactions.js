import React from 'react';
import PropTypes from 'prop-types';
import { message } from 'antd';
import { injectIntl } from 'react-intl';
import ReduxInfiniteScroll from '../vendor/ReduxInfiniteScroll';
import Loading from '../components/Icon/Loading';
import WalletTransaction from './WalletTransaction';
import { guestUserRegex } from '../helpers/regexHelpers';
import { getTransferDetails } from '../../waivioApi/ApiClient';
import { handleLoadMoreTransactions } from './WalletHelper';
import BlocktraidsTransactionModal from './BlocktraidsTransactionModal';

import './UserWalletTransactions.less';

@injectIntl
class UserWalletTransactions extends React.Component {
  static propTypes = {
    currentUsername: PropTypes.string,
    totalVestingShares: PropTypes.string.isRequired,
    totalVestingFundSteem: PropTypes.string.isRequired,
    transactions: PropTypes.arrayOf(PropTypes.shape()),
    hasMore: PropTypes.bool,
    getMoreUserTransactionHistory: PropTypes.func,
    demoTransactions: PropTypes.arrayOf(PropTypes.shape()),
    demoHasMoreActions: PropTypes.bool.isRequired,
    user: PropTypes.shape().isRequired,
    actions: PropTypes.arrayOf(PropTypes.shape()),
    getMoreUserAccountHistory: PropTypes.func.isRequired,
    intl: PropTypes.shape({
      formatMessage: PropTypes.func.isRequired,
    }).isRequired,
    isErrorLoading: PropTypes.bool,
    operationNum: PropTypes.number,
    isloadingMoreTransactions: PropTypes.bool,
    isloadingMoreDemoTransactions: PropTypes.bool,
    isMobile: PropTypes.bool,
  };

  static defaultProps = {
    currentUsername: '',
    transactions: [],
    hasMore: false,
    getMoreUserTransactionHistory: () => {},
    demoTransactions: [],
    actions: [],
    isErrorLoading: false,
    operationNum: -1,
    isloadingMoreTransactions: false,
    isloadingMoreDemoTransactions: false,
    isMobile: false,
  };

  state = {
    isOpenDetailsModal: false,
    transferDetails: null,
  };

  getWithDrawDetails = key => {
    getTransferDetails(key)
      .then(data => {
        this.setState(() => ({
          transferDetails: data,
          isOpenDetailsModal: true,
        }));
      })
      .catch(e => message.error(e.message));
  };

  isGuestPage = () => guestUserRegex.test(this.props.user && this.props.user.name);

  toggleDetailsModal = () =>
    this.setState(prevState => ({ isOpenDetailsModal: !prevState.isOpenDetailsModal }));

  getTransactionStatus = () => {
    const { transferDetails } = this.state;

    if (transferDetails.status && transferDetails.status === 'success') {
      return transferDetails.confirmed ? transferDetails.status : 'pending';
    }

    return transferDetails.status;
  };

  handleLoadMore = () => {
    const {
      currentUsername,
      operationNum,
      isloadingMoreTransactions,
      isloadingMoreDemoTransactions,
      getMoreUserAccountHistory,
      getMoreUserTransactionHistory,
      actions,
    } = this.props;

    const values = {
      username: currentUsername,
      operationNumber: operationNum,
      isLoadingMore: isloadingMoreTransactions,
      demoIsLoadingMore: isloadingMoreDemoTransactions,
      getMoreFunction: getMoreUserTransactionHistory,
      getMoreDemoFunction: getMoreUserAccountHistory,
      transferActions: actions,
      isGuest: this.isGuestPage(),
      table: false,
    };
    return handleLoadMoreTransactions(values);
  };

  render() {
    const {
      currentUsername,
      totalVestingShares,
      totalVestingFundSteem,
      transactions,
      hasMore,
      demoHasMoreActions,
      demoTransactions,
      intl,
      isErrorLoading,
      isMobile,
    } = this.props;
    const { isOpenDetailsModal, transferDetails } = this.state;

    return (
      <React.Fragment>
        <div className="UserWalletTransactions">
          <ReduxInfiniteScroll
            loadMore={this.handleLoadMore}
            hasMore={this.isGuestPage() ? demoHasMoreActions : hasMore}
            elementIsScrollable={false}
            threshold={500}
            loader={
              !isErrorLoading && (
                <div className="UserWalletTransactions__loader">
                  <Loading />
                </div>
              )
            }
          >
            <div />
            {this.isGuestPage()
              ? demoTransactions.map(demoTransaction => (
                  <WalletTransaction
                    isGuestPage={this.isGuestPage()}
                    key={`${demoTransaction.trx_id}${demoTransaction.actionCount}`}
                    transaction={demoTransaction}
                    currentUsername={currentUsername}
                    totalVestingShares={totalVestingShares}
                    totalVestingFundSteem={totalVestingFundSteem}
                    handleDetailsClick={this.getWithDrawDetails}
                    isMobile={isMobile}
                  />
                ))
              : transactions.map(transaction => (
                  <WalletTransaction
                    isGuestPage={this.isGuestPage()}
                    key={`${transaction.timestamp}${transaction.operationNum}`}
                    transaction={transaction}
                    currentUsername={currentUsername}
                    totalVestingShares={totalVestingShares}
                    totalVestingFundSteem={totalVestingFundSteem}
                    handleDetailsClick={this.getWithDrawDetails}
                    isMobile={isMobile}
                  />
                ))}
          </ReduxInfiniteScroll>
        </div>
        {transferDetails && (
          <BlocktraidsTransactionModal
            isOpenDetailsModal={isOpenDetailsModal}
            transferDetails={transferDetails}
            intl={intl}
            toggleDetailsModal={this.toggleDetailsModal}
            status={this.getTransactionStatus()}
          />
        )}
      </React.Fragment>
    );
  }
}

export default UserWalletTransactions;
