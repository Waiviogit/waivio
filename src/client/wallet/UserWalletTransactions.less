@import (reference) '../styles/custom.less';
@import (reference) '../styles/modules/variables.less';

.UserWalletTransactions {
  background-color: @white;
  border-right: 0;
  border-left: 0;
  border-radius: 0;
  margin: 20px 0;

  &:empty {
    display: none;
  }

  &__svg {
    height: 40px;
    width: 40px;
    margin-top: 5px;
  }
  &__amount {
    color: @persian-red;
  }
  &__completed {
    color: @green;
  }

  &__icon-wrapper {
    align-items: center;
    color: @blue-heather;
    background-color: @blue-solitude;
    border-radius: 4px;
    display: flex;
    height: 40px;
    justify-content: center;
    margin-top: 4px;
    width: 60px;
  }

  &__loader {
    border-bottom: @base-border;
    padding: 20px;

    &:nth-of-type(2) {
      border-top: @base-border;
    }

    @media @small {
      border-right: @base-border;
      border-left: @base-border;
      border-bottom-left-radius: @border-radius-base;
      border-bottom-right-radius: @border-radius-base;

      &:nth-of-type(2) {
        border-top-left-radius: @border-radius-base;
        border-top-right-radius: @border-radius-base;
      }
    }
  }

  &__transaction {
    color: @grey-nightRider;
    display: flex;
    padding: 10px 15px;
    border-bottom: @base-border;

    &:nth-of-type(2) {
      border-top: @base-border;
    }

    @media @small {
      border-right: @base-border;
      border-left: @base-border;

      &:nth-of-type(2),
      &:last-of-type {
        border-top-left-radius: @border-radius-base;
        border-top-right-radius: @border-radius-base;
      }
    }
  }

  &__memo {
    word-wrap: break-word;
  }

  &__avatar {
    margin-top: 5px;
  }

  &__icon-container {
    align-items: center;
    color: @blue-heather;
    background-color: @blue-solitude;
    border-radius: 4px;
    display: flex;
    height: 40px;
    justify-content: center;
    margin-top: 4px;
    width: 44px;
  }

  &__icon {
    font-size: 22px !important;
  }

  &__content {
    display: flex;
    flex-direction: column;
    padding-left: 10px;
    width: 100%;
    overflow: hidden;

    & span.username {
      transition: 0.5s;

      &:hover {
        color: @orange-hot;
      }
    }
  }

  &__content-recipient {
    display: block;
    white-space: nowrap;
    color: @grey-mortar;

    @media @xsmall {
      display: flex;
      flex-direction: row;
    }

    span:first-child {
      white-space: break-spaces;
    }
  }

  &__timestamp {
    color: @blue-botticelli;
    font-size: 14px;
  }

  &__transfer {
    color: @red;
    margin-left: auto;
  }

  &__marginLeft {
    color: @black;
    margin-left: auto;
  }

  &__received {
    color: @green;
    margin-left: auto;

    &-self {
      color: @black-text;
    }
  }

  &__limit-order {
    color: @black;
    margin-left: auto;
  }

  &__cancel-order-current-pays {
    color: @black;
    margin-left: auto;
  }

  .cancel-order-open-pays {
    color: @black;
  }

  &__received.guest {
    color: @grey-mortar;
    margin-left: auto;
  }

  &__payout {
    color: @green;
    margin-left: auto;
  }

  &__payout-black {
    color: @black;
    margin-left: auto;
  }

  &__payout-rewards {
    &:before {
      content: ' + ';
      color: @green;
    }

    &:last-of-type {
      &:after {
        content: '';
      }
    }
  }

  @media @small {
    border-radius: @border-radius-base;
  }

  &__modal-row {
    &:first-child {
      padding: 0 0 20px;
    }

    &:last-child {
      border: none;
      padding: 12px 0 0;
    }

    padding: 12px 0 20px;
    border-bottom: 1px solid #e8e8e8;
  }

  &__modal-subtitle {
    font-weight: bold;
  }

  &__amountColor {
    &--black {
      color: @black;
    }

    &--red {
      color: @red;
    }

    &--green {
      color: @green;
    }
  }
}

.arrows {
  font-style: initial;
}

div[data-dir='rtl'] .UserWalletTransactions {
  &__content {
    padding-left: 0;
    padding-right: 10px;
  }

  &__payout {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }

  &__transfer {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }

  &__received {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }
}

@import 'UserWalletTransactions-nightmode';
