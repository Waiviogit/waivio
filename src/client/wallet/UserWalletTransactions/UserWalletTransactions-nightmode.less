@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/modules/variables.less';

.nightmode {
  .UserWalletTransactions {
    background-color: @grey-nero;
    border: 1px solid @normal-grey;

    &__delegated {
      &-color {
        color: @white-smoke;
      }
    }

    &__received {
      &-self {
        color: @green;
      }
    }

    &__loader {
      border-bottom: @border-width-base @border-style-base @normal-grey;

      &:nth-of-type(2) {
        border-top: @border-width-base @border-style-base @normal-grey;
      }

      @media @small {
        border-right: @border-width-base @border-style-base @normal-grey;
        border-left: @border-width-base @border-style-base @normal-grey;
      }
    }

    &__cancel-order-current-pays {
      color: @white;
    }

    .cancel-order-open-pays {
      color: @white;
    }

    &__transaction {
      color: @white-gainsboro;
      border-bottom: @border-width-base @border-style-base @normal-grey;
    }

    &__limit-order {
      color: @white;
    }

    &__icon-container {
      color: @blue-heather;
      background-color: @blue-solitude;
    }

    &__content-recipient {
      color: @white-smoke;
    }

    &__marginLeft {
      color: @white-smoke;
    }

    &__payout-black {
      color: @white-smoke;
    }

    &__amount {
      &--red {
        color: @red;
      }

      &--green {
        color: @green;
      }
    }
  }
}
