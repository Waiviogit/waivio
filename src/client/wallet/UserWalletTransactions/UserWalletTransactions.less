@import (reference) '../../styles/custom.less';
@import (reference) '../../styles/modules/variables.less';

.UserWalletTransactions {
  background-color: @white;
  border-radius: 4px;
  border: @base-border;
  margin: 20px 0;

  &__delegated {
    font-weight: 500;
    margin-left: 4px;

    &-color {
      color: @black;
    }

    &-text {
      font-weight: lighter;
      color: @dim-gray;
    }
  }

  &:empty {
    display: none;
  }

  &__convert-icon {
    height: 20px;
    width: 20px;
  }

  &__request {
    color: @red;
  }

  &__completed {
    color: @green;
  }

  &__icon-wrapper {
    align-items: center;
    color: @blue-heather;
    background-color: @blue-solitude;
    border-radius: 4px;
    display: flex;
    height: 40px;
    justify-content: center;
    margin-top: 4px;
    width: 60px;
  }

  &__loader {
    border-bottom: @base-border;
    padding: 20px;

    &:nth-of-type(2) {
      border-top: @base-border;
    }

    @media @small {
      border-right: @base-border;
      border-left: @base-border;
      border-bottom-left-radius: @border-radius-base;
      border-bottom-right-radius: @border-radius-base;

      &:nth-of-type(2) {
        border-top-left-radius: @border-radius-base;
        border-top-right-radius: @border-radius-base;
      }
    }
  }

  &__transaction {
    color: @grey-nightRider;
    display: flex;
    padding: 10px 15px;
    border-bottom: @base-border;

    @media @small {
      &:nth-of-type(2),
      &:last-of-type {
        border-top-left-radius: @border-radius-base;
        border-top-right-radius: @border-radius-base;
      }
    }
  }

  &__memo {
    word-wrap: break-word;
  }

  &__avatar {
    margin-top: 5px;
  }

  &__icon-container {
    align-items: center;
    color: @blue-heather;
    background-color: @blue-solitude;
    border-radius: 4px;
    display: flex;
    height: 40px;
    justify-content: center;
    margin-top: 4px;
    width: 44px;
  }

  &__icon {
    font-size: 22px;
  }

  &__content {
    display: flex;
    flex-direction: column;
    padding-left: 10px;
    width: 100%;
    overflow: hidden;

    & span.username {
      transition: 0.5s;

      &:hover {
        color: @orange-hot;
      }
    }
  }

  &__content-recipient {
    display: block;
    white-space: nowrap;
    color: @grey-mortar;

    @media @xsmall {
      display: flex;
      flex-direction: row;
    }

    span:first-child {
      white-space: break-spaces;
    }
  }

  &__timestamp {
    color: @blue-botticelli;
    font-size: 14px;
  }

  &__transfer {
    color: @red;
    margin-left: auto;
  }

  &__marginLeft {
    color: @black;
    margin-left: auto;

    &--green {
      color: @green;
    }

    &--red {
      color: @red;
    }
  }

  &__received {
    color: @green;
    margin-left: auto;

    &-self {
      color: @black-text;
    }
  }

  &__sent {
    color: @red;
    margin-left: auto;
  }

  &__limit-order {
    color: @black;
    margin-left: auto;
  }

  &__cancel-order-current-pays {
    color: @black;
    margin-left: auto;
  }

  .cancel-order-open-pays {
    color: @black;
  }

  &__received.guest {
    color: @grey-mortar;
    margin-left: auto;
  }

  &__payout {
    color: @green;
    margin-left: auto;
  }

  &__payout-black {
    color: @black;
    margin-left: auto;
  }

  &__payout-rewards {
    &:before {
      content: ' + ';
      color: @green;
    }

    &:last-of-type {
      &:after {
        content: '';
      }
    }
  }

  @media @small {
    border-radius: @border-radius-base;
  }

  &__modal-row {
    &:first-child {
      padding: 0 0 20px;
    }

    &:last-child {
      border: none;
      padding: 12px 0 0;
    }

    padding: 12px 0 20px;
    border-bottom: 1px solid #e8e8e8;
  }

  &__modal-subtitle {
    font-weight: bold;
  }

  &__amountColor {
    &--black {
      color: @black;
    }

    &--red {
      color: @red;
    }

    &--green {
      color: @green;
    }
  }

  &__amount {
    &--red {
      color: @red;
    }

    &--green {
      color: @green;
    }
  }
}

.arrows {
  font-style: initial;
}

div[data-dir='rtl'] .UserWalletTransactions {
  &__content {
    padding-left: 0;
    padding-right: 10px;
  }

  &__payout {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }

  &__transfer {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }

  &__received {
    direction: ltr;
    margin-right: auto;
    margin-left: 0;
  }
}

@import './UserWalletTransactions-nightmode';
