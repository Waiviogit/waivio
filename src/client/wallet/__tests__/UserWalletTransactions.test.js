import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import configureStore from 'redux-mock-store';
import UserWalletTransactions from '../UserWalletTransactions';

describe('(Component) UserWalletTransactions', () => {
  it('renders and matches snapshot', () => {
    const mockStore = configureStore();
    const props = {
      transactions: [
        {
          timestamp: '0',
          op: [
            'transfer_to_vesting',
            {
              amount: '100 HIVE',
            },
          ],
        },
        {
          timestamp: '0',
          op: [
            'transfer',
            {
              from: 'hellosteem1',
              memo: 'transfer memo',
              amount: '100 HIVE',
            },
          ],
        },
      ],
      currentUsername: 'hellosteem',
      totalVestingShares: '0',
      totalVestingFundSteem: '0',
      loadingMoreUsersAccountHistory: false,
      userHasMoreActions: false,
      getMoreUserAccountHistory: () => {},
    };
    const wrapper = shallow(
      <Provider store={mockStore()}>
        <IntlProvider locale="en">
          <UserWalletTransactions {...props} />
        </IntlProvider>
        ,
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
