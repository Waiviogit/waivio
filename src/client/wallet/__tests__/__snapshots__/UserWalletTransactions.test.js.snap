// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) UserWalletTransactions renders and matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <IntlProvider
      locale="en"
    >
      <InjectIntl(UserWalletTransactions)
        currentUsername="hellosteem"
        getMoreUserAccountHistory={[Function]}
        loadingMoreUsersAccountHistory={false}
        totalVestingFundSteem="0"
        totalVestingShares="0"
        transactions={
          Array [
            Object {
              "op": Array [
                "transfer_to_vesting",
                Object {
                  "amount": "100 HIVE",
                },
              ],
              "timestamp": "0",
            },
            Object {
              "op": Array [
                "transfer",
                Object {
                  "amount": "100 HIVE",
                  "from": "hellosteem1",
                  "memo": "transfer memo",
                },
              ],
              "timestamp": "0",
            },
          ]
        }
        userHasMoreActions={false}
      />
    </IntlProvider>
    ,
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <IntlProvider
          locale="en"
        >
          <InjectIntl(UserWalletTransactions)
            currentUsername="hellosteem"
            getMoreUserAccountHistory={[Function]}
            loadingMoreUsersAccountHistory={false}
            totalVestingFundSteem="0"
            totalVestingShares="0"
            transactions={
              Array [
                Object {
                  "op": Array [
                    "transfer_to_vesting",
                    Object {
                      "amount": "100 HIVE",
                    },
                  ],
                  "timestamp": "0",
                },
                Object {
                  "op": Array [
                    "transfer",
                    Object {
                      "amount": "100 HIVE",
                      "from": "hellosteem1",
                      "memo": "transfer memo",
                    },
                  ],
                  "timestamp": "0",
                },
              ]
            }
            userHasMoreActions={false}
          />
        </IntlProvider>,
        ",",
      ],
      "value": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <InjectIntl(UserWalletTransactions)
            currentUsername="hellosteem"
            getMoreUserAccountHistory={[Function]}
            loadingMoreUsersAccountHistory={false}
            totalVestingFundSteem="0"
            totalVestingShares="0"
            transactions={
              Array [
                Object {
                  "op": Array [
                    "transfer_to_vesting",
                    Object {
                      "amount": "100 HIVE",
                    },
                  ],
                  "timestamp": "0",
                },
                Object {
                  "op": Array [
                    "transfer",
                    Object {
                      "amount": "100 HIVE",
                      "from": "hellosteem1",
                      "memo": "transfer memo",
                    },
                  ],
                  "timestamp": "0",
                },
              ]
            }
            userHasMoreActions={false}
          />,
          "locale": "en",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "currentUsername": "hellosteem",
            "getMoreUserAccountHistory": [Function],
            "loadingMoreUsersAccountHistory": false,
            "totalVestingFundSteem": "0",
            "totalVestingShares": "0",
            "transactions": Array [
              Object {
                "op": Array [
                  "transfer_to_vesting",
                  Object {
                    "amount": "100 HIVE",
                  },
                ],
                "timestamp": "0",
              },
              Object {
                "op": Array [
                  "transfer",
                  Object {
                    "amount": "100 HIVE",
                    "from": "hellosteem1",
                    "memo": "transfer memo",
                  },
                ],
                "timestamp": "0",
              },
            ],
            "userHasMoreActions": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      ",",
    ],
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
        "displayName": "ReactRedux",
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <IntlProvider
            locale="en"
          >
            <InjectIntl(UserWalletTransactions)
              currentUsername="hellosteem"
              getMoreUserAccountHistory={[Function]}
              loadingMoreUsersAccountHistory={false}
              totalVestingFundSteem="0"
              totalVestingShares="0"
              transactions={
                Array [
                  Object {
                    "op": Array [
                      "transfer_to_vesting",
                      Object {
                        "amount": "100 HIVE",
                      },
                    ],
                    "timestamp": "0",
                  },
                  Object {
                    "op": Array [
                      "transfer",
                      Object {
                        "amount": "100 HIVE",
                        "from": "hellosteem1",
                        "memo": "transfer memo",
                      },
                    ],
                    "timestamp": "0",
                  },
                ]
              }
              userHasMoreActions={false}
            />
          </IntlProvider>,
          ",",
        ],
        "value": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "subscription": Subscription {
            "handleChangeWrapper": [Function],
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          },
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <InjectIntl(UserWalletTransactions)
              currentUsername="hellosteem"
              getMoreUserAccountHistory={[Function]}
              loadingMoreUsersAccountHistory={false}
              totalVestingFundSteem="0"
              totalVestingShares="0"
              transactions={
                Array [
                  Object {
                    "op": Array [
                      "transfer_to_vesting",
                      Object {
                        "amount": "100 HIVE",
                      },
                    ],
                    "timestamp": "0",
                  },
                  Object {
                    "op": Array [
                      "transfer",
                      Object {
                        "amount": "100 HIVE",
                        "from": "hellosteem1",
                        "memo": "transfer memo",
                      },
                    ],
                    "timestamp": "0",
                  },
                ]
              }
              userHasMoreActions={false}
            />,
            "locale": "en",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "currentUsername": "hellosteem",
              "getMoreUserAccountHistory": [Function],
              "loadingMoreUsersAccountHistory": false,
              "totalVestingFundSteem": "0",
              "totalVestingShares": "0",
              "transactions": Array [
                Object {
                  "op": Array [
                    "transfer_to_vesting",
                    Object {
                      "amount": "100 HIVE",
                    },
                  ],
                  "timestamp": "0",
                },
                Object {
                  "op": Array [
                    "transfer",
                    Object {
                      "amount": "100 HIVE",
                      "from": "hellosteem1",
                      "memo": "transfer memo",
                    },
                  ],
                  "timestamp": "0",
                },
              ],
              "userHasMoreActions": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        ",",
      ],
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
          "displayName": "ReactRedux",
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
